app/controlplane/pkg/data/membership.go

```diff
@@ -289,12 +289,12 @@ func (r *MembershipRepo) Delete(ctx context.Context, id uuid.UUID) error {
 			}
 
 			// Decrement the member count of each group this user was a member of
-			// We don't need a separate check for groups the user was a maintainer of
-			// because all memberships were already deleted above
+			// Only decrement if member_count > 0 to avoid negative values
 			if _, err := tx.Group.Update().
 				Where(
-					group.HasMembersWith(user.ID(userID)),
+					group.HasGroupUsersWith(groupmembership.UserID(userID), groupmembership.DeletedAtIsNil()),
 					group.HasOrganizationWith(organization.ID(orgID)),
+					group.MemberCountGT(0),
```

Comment:
I think this is wrong, it's not guarantee that the computation is correct does it?